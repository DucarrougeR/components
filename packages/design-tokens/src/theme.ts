import { InterpolationValue } from 'styled-components'
import {
  defaultReset,
  Easings,
  FontFamilyChoices,
  FontSizeRamp,
  FontWeightRamp,
  LineHeightRamp,
  Palette,
  Radii,
  SemanticColors,
  Shadows,
  SpaceRamp,
  Transitions,
} from './system'

/**
 * Theme attributes shouldn't be exported as they should be consumed via `theme` rather than via
 * direct import.
 */
import {
  breakpoints,
  easings,
  fontFamilies,
  fontSizes,
  fontWeights,
  lineHeights,
  palette,
  radii,
  semanticColors,
  shadows,
  space,
  transitions,
} from './tokens'

export interface Theme {
  breakpoints: string[]
  colors: { palette: Palette; semanticColors: SemanticColors }
  easings: Easings
  fontSizes: FontSizeRamp
  fontWeights: FontWeightRamp
  fonts: FontFamilyChoices
  lineHeights: LineHeightRamp
  radii: Radii
  /**
   * A function that can be overridden to return different reset css properties
   * or null to remove all resets. Most base elements in Lens implement the reset.
   */
  reset: () => InterpolationValue
  shadows: Shadows
  space: SpaceRamp
  transitions: Transitions
}

export const theme: Theme = {
  breakpoints,
  colors: { palette, semanticColors },
  easings,
  fontSizes,
  fontWeights,
  fonts: fontFamilies,
  lineHeights,
  radii,
  reset: defaultReset,
  shadows,
  space,
  transitions,
}
