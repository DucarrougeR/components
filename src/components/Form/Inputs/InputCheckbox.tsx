// Typescript component boilerplate for generating meaningful declaration files.
import * as React from 'react'
import styled, { StyledComponentClass } from '../../../styled_components'
export { StyledComponentClass }
import { ThemeInterface } from '../../../themes'
export { ThemeInterface }
import { themeSpacing } from '../../../themes/theme_spacing'
// End Typescript component boilerplate

export const InputCheckbox = styled.input.attrs({
  type: 'checkbox'
})`
  padding: 0 ${themeSpacing.s};
`
