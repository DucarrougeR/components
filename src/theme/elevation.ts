export type ElevationLayers = '1' | '2' | '3' | '4' | '5' | '6' | '7' | 'max'

export type Shadows = { [K in ElevationLayers]?: string }

export const shadows: Shadows = {
  1: '0 1px 2px rgba(0, 0, 0, 0.1), 0px 1px 1px rgba(0, 0, 0, 0.05)',
  2: '0 3px 4px rgba(0, 0, 0, 0.11), 0 1px 2px rgba(0, 0, 0, 0.04)',
  3: '0 6px 10px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.04)',
  4: '0 9px 18px rgba(0, 0, 0, 0.14), 0 1px 2px rgba(0, 0, 0, 0.04),  0 10px 8px 4px rgba(0, 0, 0, 0.01)',
  5: '0 14px 20px rgba(0, 0, 0,.15), 0 1px 1px rgba(0, 0, 0, 0.04),  0 14px 10px 8px rgba(0, 0, 0, 0.02)',
  6: '0 22px 42px 0px rgba(0, 0, 0,.18),  0 1px 1px rgba(0, 0, 0, 0.05), 0 16px 20px 10px rgba(0, 0, 0, 0.025)',
}
