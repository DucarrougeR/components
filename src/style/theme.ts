import { rem } from 'polished'
import { InterpolationValue } from 'styled-components'
import {
  SizeLarge,
  SizeMedium,
  SizeNone,
  SizeSmall,
  SizeXLarge,
  SizeXSmall,
  SizeXXLarge,
  SizeXXXLarge,
} from '../types'
import { Easings, easings } from './easings'
import { shadows, Shadows } from './elevation'
import { fontFaces, FontFaces } from './font_faces'
import { FontRamp, fontSizes, lineHeights } from './font_sizes'
import { fontWeights } from './font_weights'
import { palette, Palette } from './palette'
import { SemanticColors, semanticColors } from './semantic_colors'
import { transitions, Transitions } from './transitions'

export * from './easings'
export * from './elevation'
export * from './font_sizes'
export * from './palette'
export * from './semantic_colors'
export * from './transitions'

export type SpacingSizes =
  | SizeNone
  | SizeLarge
  | SizeMedium
  | SizeSmall
  | SizeXLarge
  | SizeXSmall
  | SizeXXLarge
  | SizeXXXLarge

export interface Theme {
  breakpoints: string[]
  easings: Easings
  fontFaces: FontFaces
  fontSizes: FontRamp
  fontWeights: Record<string, number>
  lineHeights: FontRamp
  palette: Palette
  /**
   * A function that can be overridden to return different reset css properties
   * or null to remove all resets. Most base elements in Lens implement the reset.
   */
  reset: () => InterpolationValue
  semanticColors: SemanticColors
  shadows: Shadows
  space: Record<SpacingSizes, string>
  transitions: Transitions
}

export const theme: Theme = {
  breakpoints: ['30rem', '48rem', '64rem', '75rem', '90rem'],
  easings,
  fontFaces,
  fontSizes,
  fontWeights,
  lineHeights,
  palette,
  reset: () => {
    return {
      border: 0,
      font: 'inherit',
      fontSize: '100%',
      margin: 0,
      padding: 0,
      verticalAlign: 'baseline',
    }
  },
  semanticColors,
  shadows,
  space: {
    // tslint:disable:object-literal-sort-keys
    none: 'none',
    xsmall: rem(4),
    small: rem(8),
    medium: rem(16),
    large: rem(20),
    xlarge: rem(32),
    xxlarge: rem(40),
    xxxlarge: rem(60),
    // tslint:enable:object-literal-sort-keys
  },
  transitions,
}
