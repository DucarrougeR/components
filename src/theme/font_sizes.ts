import { rem } from 'polished'
import { RampSizes } from '../style/ramp_sizes'

export type FontRamp = { [K in RampSizes]: string }

// tslint:disable:object-literal-sort-keys
export const fontRamp: FontRamp = {
  d1: rem(58),
  d2: rem(46),
  d3: rem(36),
  1: rem(25),
  2: rem(22),
  3: rem(18),
  4: rem(16),
  5: rem(14),
  6: rem(12),
}

// tslint:disable:object-literal-sort-keys
export const lineHeightRamp: FontRamp = {
  d1: rem(84),
  d2: rem(64),
  d3: rem(52),
  1: rem(40),
  2: rem(32),
  3: rem(28),
  4: rem(24),
  5: rem(20),
  6: rem(16),
}
