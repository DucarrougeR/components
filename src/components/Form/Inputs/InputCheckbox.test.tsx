import 'jest-styled-components'
import * as React from 'react'
import { create } from 'react-test-renderer'
import theme from '../../../themes'
import { InputCheckbox } from './InputCheckbox'

test('An InputCheckbox', () => {
  const component = create(
    <InputCheckbox name="thumbsUp" id="thumbs-up" theme={theme} />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
