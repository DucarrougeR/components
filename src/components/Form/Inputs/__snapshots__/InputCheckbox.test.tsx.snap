// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`An InputCheckbox 1`] = `
.c0 {
  padding: 0 0.5rem;
}

<input
  className="c0"
  id="thumbs-up"
  name="thumbsUp"
  type="checkbox"
/>
`;
